### PR TITLE
v2.3.0 release prep

### PR DESCRIPTION
**Pull request by shuchitak**
- Update to Jenkins shared library v0.34.0, xmosdoc v6.1.0 and use the BuildDocs() function for building documentation.
- Added cognidox part number in settings.yml
- Build the README pdf.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,33 +1,42 @@
...
+* `/doc/ <https://github.com/xmos/lib_xcore_math/tree/develop/doc>`_ - Sphinx library documentation source and build directory.
+* `/examples/ <https://github.com/xmos/lib_xcore_math/tree/develop/examples>`__ - Example applications for ``lib_xcore_math``.
+* `/tests/ <https://github.com/xmos/lib_xcore_math/tree/develop/tests>`__ - Unit test projects for ``lib_xcore_math``.
 
 
 Requirements
-------------
+============
 
 * xTIMEcomposer tools version 15.0.6 or later
```
Comment (ed-xmos) [inline README.rst:41]: 15.3.0

File: README.rst [hunk 2]
```diff
@@ -1,33 +1,42 @@
...
 * CMake (for building the library and unit tests, version 3.14 or later)
```
Comment (ed-xmos) [inline README.rst:42]: I think we are specifying 3.21 for cmake now

**Review by ed-xmos — APPROVED** [review]
Looks good - Approval subject to version comments made

File: README.rst (hunk 1 above)
Comment (shuchitak) [reply README.rst:41]: Thanks! will fix

**Pull request merged**